libraries/lib-music-information-retrieval/MirUtils.cpp

```diff
@@ -0,0 +1,80 @@
+#include "MirUtils.h"
+
+#include <cmath>
+#include <numeric>
+
+namespace MIR
+{
+namespace
+{
+constexpr auto bpmStdDev = 29.7953;
+constexpr auto pi = 3.141592653589793;
+
+constexpr bool IsPrimeDecompositionTwoThreeOnly(int n)
+{
+   while (n % 2 == 0)
+      n /= 2;
+   while (n % 3 == 0)
+      n /= 3;
+   return n == 1;
+}
+
+constexpr int CountThreesInPrimeDecomposition(int n)
+{
+   int count = 0;
+   while (n % 3 == 0)
+   {
+      ++count;
+      n /= 3;
+   }
+   return count;
+}
+
+// Function to generate numbers whose prime factorization contains only twos or
+// threes
+std::vector<int> GetPowersOf2And3(int lower, int upper)
```

Comment:
The function name doesn't suggest exactly what it does.  6 is not a power of 2 or of 3.